src/XModem.h

```diff
@@ -90,11 +94,15 @@ class XModem {
     void build_packet(struct packet *p, byte *id, byte *data, size_t data_len);
     bool send_packet(struct packet *p);
     bool close_tx();
-
+    
     void increment_id(byte *id, size_t length);
     byte tx_signal(byte signal);
     byte rx_signal();
     bool find_byte_timed(byte b, byte timeout_secs);
+    
+    
 };
 
+extern XModem myXModem;
```

Comment:
From what I understand you shouldn't need to export an object instance from within the library code, furthermore from within the arduino platform perspective this has the drawback that developers using the library would use it like a singleton instead of creating as many instances as they need. For example if the hardware has the right kind of hardware you could run 2 independent instances over 2 separate serial communication channels (assuming the micro-controller can keep up of coarse)